efficientdet/keras/postprocess.py

```diff
@@ -238,6 +243,133 @@ def postprocess_combined(params, cls_outputs, box_outputs, image_scales=None):
+
+  # TODO(b/175166514): Consider computing Top-K boxes & anchors here. We don't
+  # do this currently since the resultant graph does not support TFLite
+  # delegates well. `topk_class_boxes` won't work as-is, since the outputs
+  # will need to be modified appropriately for TFLite op's consumption.
+
+  # TFLite's object detection APIs require normalized anchors.
+  height, width = utils.parse_image_size(params['image_size'])
+  normalize_factor = tf.constant([height, width, height, width],
+                                 dtype=tf.float32)
+  normalized_anchors = eval_anchors.boxes / normalize_factor
+  decoded_anchors = anchors.decode_anchors_to_centersize(
+      box_outputs, normalized_anchors)
+
+  # convert logits to scores.
+  scores = tf.math.sigmoid(cls_outputs)
+
+  return box_outputs, scores, decoded_anchors
+
+
+def postprocess_tflite(params, cls_outputs, box_outputs):
+  """Post processing for conversion to TFLite.
+
+  Mathematically same as postprocess_global, except that the last portion of the
+  TF graph constitutes a dummy `tf.function` that contains an annotation for
+  conversion to TFLite's custom NMS op. Using this custom op allows features
+  like post-training quantization & accelerator support.
+  NOTE: This function does NOT return a valid output, and is only meant to
+  generate a SavedModel for TFLite conversion via MLIR.
+  For TFLite op details, see tensorflow/lite/kernels/detection_postprocess.cc
+
+  Args:
+    params: a dict of parameters.
+    cls_outputs: a list of tensors for classes, each tensor denotes a level of
+      logits with shape [1, H, W, num_class * num_anchors].
+    box_outputs: a list of tensors for boxes, each tensor ddenotes a level of
+      boxes with shape [1, H, W, 4 * num_anchors]. Each box format is [y_min,
+      x_min, y_max, x_man].
+
+  Returns:
+    A (dummy) tuple of (boxes, scores, classess, valid_len).
+  """
+  box_outputs, scores, decoded_anchors = tflite_pre_nms(params, cls_outputs,
+                                                        box_outputs)
+
+  # There is no TF equivalent for TFLite's custom post-processing op.
+  # So we add an 'empty' composite function here, that is legalized to the
+  # custom op with MLIR.
+  # For details, see:
+  # tensorflow/compiler/mlir/lite/utils/nms_utils.cc
+  @tf.function(experimental_implements=tflite_nms_implements_signature(params))
+  # pylint: disable=g-unused-argument,unused-argument
+  def dummy_post_processing(box_encodings, class_predictions, anchor_boxes):
+    boxes = tf.constant(0.0, dtype=tf.float32, name='boxes')
+    scores = tf.constant(0.0, dtype=tf.float32, name='scores')
+    classes = tf.constant(0.0, dtype=tf.float32, name='classes')
+    num_detections = tf.constant(0.0, dtype=tf.float32, name='num_detections')
+    return boxes, classes, scores, num_detections
+
+  return dummy_post_processing(box_outputs, scores, decoded_anchors)[::-1]
```

Comment:
Why reverse results?

Reply:
Sorry, what do you mean?

Reply:
[::-1] reverses the results.
Is it necessary?

Reply:
Ah, good point. This is just a placeholder, the reverse might not be necessary.
Since tflite team added this, I will double check with them later.